ingenialink/servo.py

```diff
@@ -853,22 +853,33 @@ def monitoring_remove_data(self) -> None:
         self.write(self.MONITORING_REMOVE_DATA, data=1, subnode=0)
 
     def monitoring_set_mapped_register(
-        self, channel: int, address: int, subnode: int, dtype: int, size: int
+        self, channel: int, uid: str, size: int, axis: Optional[int] = None
     ) -> None:
         """Set monitoring mapped register.
 
         Args:
             channel: Identity channel number.
-            address: Register address to map.
-            subnode: Subnode to be targeted.
-            dtype: Register data type.
+            uid: Register uid.
             size: Size of data in bytes.
+            axis: axis. Should be specified if multiaxis, None otherwise.
 
+        Raises:
+            RuntimeError: if the register is not monitoreable.
         """
+        register = self.dictionary.get_register(uid, axis=axis)
+        if not register.is_monitoreable:
+            raise RuntimeError("Register is not monitoreable.")
```

Comment:
add uid to error?. @GullinBustin Is going to be shown in ML3? or there are protections first?